custom-targets/vertex-ai/model-deployer/vertexai.go

```diff
@@ -0,0 +1,199 @@
+		deployedModels[modelNameWithVersion] = dm
+	}
+
+	delete(deployedModels, currentModel)
+
+	if len(deployedModels) != 1 {
+		return "", fmt.Errorf("unable to resolve previous deployed currentModel to canary against. Not including the current currentModel to be deployed, the endpoint has %d deployed models but expected only one", len(deployedModels))
+	}
+
+	firstModel := []*aiplatform.GoogleCloudAiplatformV1DeployedModel{}
+
+	for _, dm := range deployedModels {
+		firstModel = append(firstModel, dm)
+	}
+	return firstModel[0].Id, nil
+}
+
+// resolveDeployedModelNameWithVersion returns the model resource name associated with the  provided DeployedModel
+// with its version ID attached.
+func resolveDeployedModelNameWithVersion(deployedModel *aiplatform.GoogleCloudAiplatformV1DeployedModel) string {
+	if strings.Contains(deployedModel.Model, "@") {
+		return deployedModel.Model
+	}
+	return fmt.Sprintf("%s@%s", deployedModel.Model, deployedModel.ModelVersionId)
+}
+
+// resolveModelWithVersion returns the model resource name its version ID attached.
+func resolveModelWithVersion(model *aiplatform.GoogleCloudAiplatformV1Model) string {
+	if strings.Contains(model.Name, "@") {
+		return model.Name
+	}
+	return fmt.Sprintf("%s@%s", model.Name, model.VersionId)
+}
+
+// extracts the region from the model region name.
+func fetchRegionFromModel(modelName string) (string, error) {
+	matches := modelRegex.FindStringSubmatch(modelName)
+	if len(matches) == 0 {
+		return "", fmt.Errorf("unable to parse model name")
+	}
+
+	return matches[2], nil
+}
+
+// extracts the region from the endpoint resource name.
+func fetchRegionFromEndpoint(endpointName string) (string, error) {
+	matches := endpointRegex.FindStringSubmatch(endpointName)
+	if len(matches) == 0 {
+		return "", fmt.Errorf("unable to parse endpoint name")
+	}
+
+	return matches[2], nil
+}
+
+// newAIPlatformService generates a Service that can make API calls in the specified region.
+func newAIPlatformService(ctx context.Context, region string) (*aiplatform.Service, error) {
+	endPointOption := option.WithEndpoint(fmt.Sprintf("%s-aiplatform.googleapis.com", region))
+	regionalService, err := aiplatform.NewService(ctx, endPointOption)
+	if err != nil {
+		return nil, fmt.Errorf("unable to authenticate")
+	}
+
+	return regionalService, nil
+}
+
+// fetchModel calls the aiplatform API to fetch the Vertex AI model using the given model name.
+func fetchModel(service *aiplatform.Service, modelName string) (*aiplatform.GoogleCloudAiplatformV1Model, error) {
+	model, err := service.Projects.Locations.Models.Get(modelName).Do()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get model: %v", err)
+	}
+	return model, nil
+}
+
+// validateRequest performs validation on the request.
+func validateRequest(modelNameFromDeployParameter, endpointName string, minReplicaCountParameter int64, deployedModel *aiplatform.GoogleCloudAiplatformV1DeployedModel) error {
```

Comment:
The validation functions feel like they belong in `render.go` FWIW since it's not for interacting with vertex ai.

Reply:
moved methods that do validation back to `render.go`